### ISSUE TITLE
Running graphQL server on a different port (microservice)

### ISSUE THREAD

**Issue by mairh** [sysgears/apollo-fullstack-starter-kit#172]
First of all thanks for developing and maintaining the starter kit. I would like to understand how can I run the GraphQL server as a microservice.

For e.g, I have to develop both web and mobile app using react and react native. However, I would like the web and mobile app to use the same GraphQL server available over https.

Do you have any pointers on how to achieve that? What I am trying to achieve at the end is what [grahpcool](https://www.graph.cool/) guys are doing (you call the GraphQL server api endpoint in your app), which is not tightly coupled with the platform. So I can use the same GraphQL server on my web and mobile app.

**Comment by vlasenko** [comment]
To run GraphQL server produced by this starter kit simply set `PORT=zzzz` environment variable before running backend process. You can have multiple backend processes running on the same machine this way. We haven't explored the best way to implement microservices with this starter kit yet, but we have already fully decoupled features from one another, both client-side:
https://github.com/sysgears/apollo-fullstack-starter-kit/blob/master/src/client/modules/connector.js
https://github.com/sysgears/apollo-fullstack-starter-kit/blob/master/src/client/modules/index.js
and server-side:
https://github.com/sysgears/apollo-fullstack-starter-kit/blob/master/src/server/modules/connector.js
https://github.com/sysgears/apollo-fullstack-starter-kit/blob/master/src/server/modules/index.js
So the way to go with microservices would be to continue and move out these decoupled features into separate running node processes. At the moment we are still busy with other tasks for this starter kit. But if you would like to work on this and submit pull request - that would be great and we will try to help from our side.

**Comment by mairh** [comment]
Thanks for your reply. I will try to create a PR when I reach the stage of my project when I have to deploy the server on a different node process.

**Comment by vlasenko** [comment]
Sounds good, thank you @mairh !

**Comment by strefethen** [comment]
@vlasenko curious if this ever evolved beyond the discussion above. I too an interested in leveraging microservices and would like to be able to run the graphql server on a separate machine. I don't see anything in the docs mentioning it so I'm guessing the likelihood is that it didn't evolve.

**Comment by vlasenko** [comment]
@strefethen At the time this discussion started, it was pretty difficult to make microservice based arch with GraphQL. Though the kit has been modular always and it was one step away from microservice-based arch if GraphQL and Apollo allowed that. We have never bothered because we saw that GraphQL is not ready itself. Two weeks ago Apollo Federation has been released:
https://blog.apollographql.com/apollo-federation-f260cf525d21
which promises to solve microservice concerns for GraphQL.And I have created issue:
https://github.com/sysgears/apollo-universal-starter-kit/issues/1095
that states Apollo Federation should be investigated. So anyone is welcome to investigate using the kit with Apollo Federation and share his results / open pull request with Apollo Federation integration.
